src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -34,21 +31,18 @@
 class ItemDataProvider implements ItemDataProviderInterface
 {
     private $managerRegistry;
-    private $propertyNameCollectionFactory;
-    private $propertyMetadataFactory;
+    private $identifierManager;
     private $itemExtensions;
 
     /**
-     * @param ManagerRegistry                        $managerRegistry
-     * @param PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory
-     * @param PropertyMetadataFactoryInterface       $propertyMetadataFactory
-     * @param QueryItemExtensionInterface[]          $itemExtensions
+     * @param ManagerRegistry               $managerRegistry
+     * @param IdentifierManagerInterface    $identifierManager
+     * @param QueryItemExtensionInterface[] $itemExtensions
      */
-    public function __construct(ManagerRegistry $managerRegistry, PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, array $itemExtensions = [])
```

Comment:
It's because if someone initialize this class by themselves (and some people including me do) it will break.
This document is a nice reference of what is a BC break or not: http://symfony.com/doc/current/contributing/code/bc.html